mlrun/datastore/targets.py

```diff
@@ -1249,7 +1252,31 @@ def get_spark_options(self, key_column=None, timestamp_key=None, overwrite=True)
         return spark_options
 
     def purge(self):
-        pass
+        import snowflake.connector
+
+        account = self.attributes["url"].replace(".snowflakecomputing.com", "")
+        missing = [
+            key
+            for key in ["database", "db_schema", "table"]
+            if self.attributes.get(key) is None
+        ]
+        if missing:
+            raise mlrun.errors.MLRunRuntimeError(
+                f"Can't purge Snowflake target, "
+                f"some attributes are missing: {missing}"
+            )
```

Comment:
I guess `url` can be missing too? Better raise the nice error and not a `KeyError` if that happens.

Reply:
added to missing key list